internal/matrix/command/shift.go

```diff
@@ -0,0 +1,162 @@
+package command
+
+import (
+	"fmt"
+	"regexp"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/matrix-org/gomatrix"
+	"github.com/pkg/errors"
+	"github.com/snapp-incubator/matrix-on-call-bot/internal/matrix"
+	"github.com/snapp-incubator/matrix-on-call-bot/internal/matrix/message"
+	"github.com/snapp-incubator/matrix-on-call-bot/internal/model"
+)
+
+const (
+	// check if command starts with
+	// !startshift or
+	// !endshift or
+	// !listshifts.
+	startShiftParseRegexString = "^!startshift|^!endshift|^!listshifts"
+
+	minCreateShiftLength int = 2
+
+	minEndShiftLength int = 2
+)
+
+type Shift struct {
+	parseRegex *regexp.Regexp
+	shiftRepo  model.ShiftRepo
+	client     *gomatrix.Client
+}
+
+func NewShiftCmd(repo model.ShiftRepo, client *gomatrix.Client) (*Shift, error) {
+	regex := regexp.MustCompile(startShiftParseRegexString)
+
+	return &Shift{
+		parseRegex: regex,
+		shiftRepo:  repo,
+		client:     client,
+	}, nil
+}
+
+func (s *Shift) Match(message string) bool {
+	return s.parseRegex.MatchString(message)
+}
+
+func (s *Shift) Handle(event *gomatrix.Event) error {
+	raw, ok := event.Content["body"].(string)
+	if !ok {
+		return matrix.ErrInvalidBody
+	}
+
+	parts := strings.Split(raw, " ")
+
+	switch parts[0] {
+	case "!startshift":
+		return s.startShift(event, parts)
+	case "!endshift":
+		return s.endShift(event, parts)
+	case "!listshifts":
+		return s.listShifts(event)
+	}
+
+	return nil
+}
+
+func (s *Shift) startShift(event *gomatrix.Event, parts []string) error {
```

Comment:
This is the old `startShift`. Now we have `createShift` with different logic.

Reply:
Done. Please check again.